homeassistant/helpers/template.py

```diff
@@ -2785,6 +2785,32 @@ def flatten(value: Iterable[Any], levels: int | None = None) -> list[Any]:
     return flattened
 
 
+def combine(*args: Any, recursive: bool = False) -> dict[Any, Any]:
+    """Combine multiple dictionaries into one."""
+    if not args:
+        raise TypeError("combine expected at least 1 argument, got 0")
+
+    result: dict[Any, Any] = {}
+    for arg in args:
+        if not isinstance(arg, dict):
+            raise TypeError(f"combine expected a dict, got {type(arg).__name__}")
+
+        if recursive:
+            for key, value in arg.items():
+                if (
+                    key in result
+                    and isinstance(result[key], dict)
+                    and isinstance(value, dict)
+                ):
+                    result[key] = combine(result[key], value, recursive=True)
```

Comment:
I don't know if this case is interesting for templates but in Shelly we encountered a problem when the target dict key is `none` before the merge, for example:
```
{{ combine({'a': 1, 'b': none}, {'b': {'y': 2}, 'c': 4}, recursive=True) }} # b can't be assigned since it is not a dict before
```
Solution is simple if it relevant, background at https://github.com/home-assistant-libs/aioshelly/pull/808

Reply:
Added a test to show/ensure this works. Copied your example above as a test.